Lib/distutils/spawn.py

```diff
@@ -173,20 +173,28 @@ def find_executable(executable, path=None):
     os.environ['PATH'].  Returns the complete filename or None if not found.
     """
     if path is None:
-        path = os.environ.get('PATH', os.defpath)
+        path = os.environ.get('PATH', None)
+        if path is None:
+            try:
+                path = os.confstr("CS_PATH")
+            except (AttributeError, ValueError):
+                # os.confstr() or CS_PATH is not available
+                path = os.defpath
+        # bpo-35755: Don't use os.defpath if the PATH environment variable is
+        # set to an empty string to mimick Unix which command behavior
```

Comment:
I believe this is missing the `if not path: return None` that's in `shutil.which`

since right now on this branch:

```console
$ git rev-parse HEAD
077c7d7e6842121c8a7bf089b05e4a53fe124281
$ touch oops
$ chmod +x oops
$ PATH= ./python -c 'import distutils.spawn; print(distutils.spawn.find_executable("oops"))'
oops
$ PATH= ./python -c 'import shutil; print(shutil.which("oops"))'
None
```

Reply:
I move the code to handle the check in the current directory, I added an explicit "if not path: return None" with a comment to explain it, I added tests on PATH=":" and more tests in the current directory. So the behavior is now better specified by tests ;-)